whylog/config/rule.py

```diff
@@ -57,6 +64,140 @@ def get_causes_parsers(self):
     def get_effect_name(self):
         return self._effect.name
 
+    def get_search_ranges(self, effect_clues):
+        group, group_type = self._effect.get_primary_key_group()
+        if not group:
+            return self.NO_RANGE
+        parser_ranges = self._calculate_parsers_ranges(effect_clues, group, group_type)
+        if self._linkage == self.LINKAGE_OR:
+            self.fix_ranges_for_unconnected_constraints(
+                effect_clues, group, group_type, parser_ranges
+            )
+        return self._aggregate_by_log_type(parser_ranges)
+
+    def fix_ranges_for_unconnected_constraints(
+        self, effect_clues, group, group_type, parser_ranges
+    ):
+        for constraint in self._constraints:
+            if constraint['name'] not in self.DELTA_CONSTRAINTS:
+                parser_number = constraint['clues_groups'][0][0]
+                converter = CONVERTION_MAPPING[group_type]
+                primary_group_value = \
+                    effect_clues[self.get_effect_name()].regex_parameters[group - 1]
+                parser_ranges[parser_number][group_type]["left_bound"] = converter.MIN_VALUE
+                parser_ranges[parser_number][group_type]["right_bound"] = primary_group_value
+
+    def _calculate_parsers_ranges(self, effect_clues, group, group_type):
+        parser_ranges = {
+            self.EFFECT_NUMBER: self._get_effect_range(effect_clues, group, group_type)
+        }
+        queue = deque([self.EFFECT_NUMBER])
+        aggregated_constraints = self._aggregate_constraints()
+        used_parsers = set([self.EFFECT_NUMBER])
+        while queue:
+            parser_number = queue.popleft()
+            for constraint in aggregated_constraints[parser_number]:
+                clues_groups = constraint['clues_groups']
+                depended_parser_number = clues_groups[0][0]
+                base_parser_number = clues_groups[1][0]
+                if depended_parser_number in used_parsers:
+                    continue
+                if not self._is_primary_key_constraint(clues_groups):
+                    continue
+                _, group_type = self._causes[depended_parser_number - 1].get_primary_key_group()
```

Comment:
It is impossible, because earlier checks that constraint connects primary key groups